protocols/consensus-gossip/tuple-space.md

```diff
@@ -0,0 +1,192 @@
+
+Three kinds of messages are exchanged in the Tendermint algorithm: `PROPOSAL`, `PRE-VOTE`, and `PRE-COMMIT`.
+The algorithm progresses when certain conditions are satisfied over the set of messages received.
+For example, in order do decide on a value `v`, the set must include a `PROPOSAL` for `v` and `PRE-COMMIT` for the same `v` from more than two thirds of the validators for the same round.
+Since processes are subject to failures, correct processes cannot wait indefinitely for messages since the sender may be faulty.
+Hence, processes execute in rounds in which they wait for conditions to be met for sometime but, if they timeout, send negative messages that will lead to new rounds.
+
+
+## The need for Gossip Communication
+
+Progress and termination are only guaranteed in if there exists a **Global Stabilization Time (GST)** after which communication is reliable and timely (Eventual $\Delta$-Timely Communication).
+
+| Eventual $\Delta$-Timely Communication|
+|-----|
+|There is a bound $\Delta$ and an instant GST (Global Stabilization Time) such that if a correct process $p$ sends a message $m$ at a time $t \geq \text{GST}$ to a correct process $q$, then $q$ will receive $m$ before $t + \Delta$.
+
+Eventual $\Delta$-Timely Communication is used to provide **Gossip Communication**, which ensures that all messages sent by correct processes will be eventually delivered to all correct processes.
+
+|Gossip communication|
+|-----|
+| (i) If a correct process $p$ sends some message $m$ at time $t$, all correct processes will receive $m$ before $\text{max} (t,\text{GST}) + \Delta$.
+| (ii) If a correct process $p$ receives some message $m$ at time $t$, all correct processes will receive $m$ before $\text{max}(t,\text{GST}) + \Delta$.
+
+This will, in turn, lead all correct processes to eventually be able to execute a round in which the conditions to decide are met, even if only after GST is reached.
+
+Even if Eventual $\Delta$-Timely Communication is assumed, implementing Gossip Communication would be unfeasible.
+Given that all messages, even messages sent before the GST, need to be buffered to be reliably delivered between correct processes and that GST may take indefinitely long to arrive, implementing this primitive would require unbounded memory.
+
+Fortunately, while Gossip Communication is a sufficient condition for the Tendermint algorithm to terminate, it is not strictly necessary:
+i) the conditions to progress and terminate are evaluated over the messages of subsets of rounds executed, not all of them; ii) as new rounds are executed, messages in previous rounds may be become obsolete and be ignored and forgotten.
+In other words, the algorithm does not require all messages to be delivered, only messages that advance the state of the processes.
+
+## Node's state as a Tuple Space
+
+One way of looking at the information used by CometBFT nodes is as a distributed tuple space (a set of tuples) to which all nodes contribute.
```

Comment:
Not sure what the purpose of this document will be. I would add a short explanation here, why we consider this "way of looking"

Reply:
Makes sense. Added a comment in the followup PR.